### ISSUE TITLE
Windows: libcurl.dll FILE LOCKED WITH ONLY READERS

### ISSUE THREAD

**Issue by jamescoxon** [yquake2/yquake2#360]
Hi,

Firstly thanks for all the great work with yquake2, I'm part of a small team who have integrated the Nano digital currency into the game adding microtransactions for kills and payouts for the top 3 (https://github.com/Nanoquake/yquake2).

I've been working on a new build with the latest http download using curl ( and successfully installed on OS X with no problems). On Windows 10, after compiling it in the MSYS2 distribution it will only use the old quake download system.

Straight off it appeared that it was missing the libcurl.dll so I moved the libcurl-4.dll from MSYS2 to the release folder and renamed it libcurl.dll. Checking ProcMon can see that it loads the dll though it reports `FILE LOCKED WITH ONLY READERS` and then closes the file.

![success](https://user-images.githubusercontent.com/175479/51516515-a39f9c80-1e0f-11e9-8b7a-6779fc301aba.png)

stdout reports:

```------- curl initialization -------
LoadLibrary(curl.dll)
Sys_LoadLibrary failed: LoadLibrary returned 126 on curl.dll
Loading curl.dll failed!
LoadLibrary(libcurl.dll)
Sys_LoadLibrary failed: LoadLibrary returned 193 on libcurl.dll
Loading libcurl.dll failed!
```

A return 193 code apparently is `ERROR_BAD_EXE_FORMAT` though I would have thought that the libcurl.dll from the MSYS2 would be appropriate - I've tried with a number of other libcurl.dlls from other builds with no success.

I've got a bit stuck at this point, thanks in advance for the help.

**Comment by jamescoxon** [comment]
Thanks for the pointers, I don't traditionally use Windows so have forgotten about how complex dlls can get - we tried your libcurl.dll but got `Error: Modules with different CPU types were found`- I wonder whether it was 32vs64bit issue

The good news is that we've managed to get a version of libcurl.dll working using libcurl-x64.dll and dependencies from OpenSSL 1.1.1a (libcrypto-1_1-x64.dll and libssl-1_1-x64.dll). We sourced them from https://curl.haxx.se/windows/dl-7.63.0/curl-7.63.0-win64-mingw.zip and https://curl.haxx.se/windows/dl-7.63.0/openssl-1.1.1a-win64-mingw.zip. The challenge now is stopping google drive thinking they contain viruses...

Thanks again for an awesome project.
